src/components/application_manager/test/commands/mobile/register_app_interface_request_test.cc

```diff
@@ -0,0 +1,260 @@
+      kHmiLanguage;
+  ON_CALL(app_mngr_, IsHMICooperating()).WillByDefault(Return(true));
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(ApplicationSharedPtr()))
+      .WillRepeatedly(Return(mock_app));
+  ON_CALL(*mock_app, mac_address()).WillByDefault(ReturnRef(kMacAddress));
+
+  ON_CALL(app_mngr_, applications())
+      .WillByDefault(Return(DataAccessor<am::ApplicationSet>(app_set, lock_)));
+
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(ReturnRef(mock_policy_handler_));
+  ON_CALL(mock_policy_handler_, PolicyEnabled()).WillByDefault(Return(true));
+  ON_CALL(mock_policy_handler_, GetInitialAppData(_, _, _))
+      .WillByDefault(DoAll(SetHmiType(app_hmi_type_str), Return(true)));
+  ON_CALL(app_mngr_, RegisterApplication(msg_)).WillByDefault(Return(mock_app));
+
+  ON_CALL(app_mngr_, resume_controller())
+      .WillByDefault(ReturnRef(mock_resum_crt_));
+  ON_CALL(app_mngr_, connection_handler())
+      .WillByDefault(ReturnRef(mock_connection_handler_));
+  ON_CALL(mock_connection_handler_, get_session_observer())
+      .WillByDefault(ReturnRef(mock_session_observer_));
+  ON_CALL(mock_session_observer_, GetDataOnDeviceID(_, _, _, _, _))
+      .WillByDefault(Return(0));
+
+  ON_CALL(app_mngr_, hmi_capabilities())
+      .WillByDefault(ReturnRef(mock_hmi_capabilities_));
+  ON_CALL(mock_hmi_capabilities_, active_vr_language())
+      .WillByDefault(Return(kHmiLanguage));
+  ON_CALL(mock_hmi_capabilities_, active_ui_language())
+      .WillByDefault(Return(kHmiLanguage));
+  ON_CALL(*mock_app, name()).WillByDefault(ReturnRef(app_name));
+  const std::vector<uint32_t> dummy_diag_modes;
+  ON_CALL(app_mngr_settings_, supported_diag_modes())
+      .WillByDefault(ReturnRef(dummy_diag_modes));
+
+  ON_CALL(app_mngr_settings_, sdl_version())
+      .WillByDefault(ReturnRef(kDummyString));
+  ON_CALL(mock_hmi_capabilities_, ccpu_version())
+      .WillByDefault(ReturnRef(kDummyString));
+  ON_CALL(*mock_app, app_icon_path()).WillByDefault(ReturnRef(kDummyString));
+  ON_CALL(*mock_app, language()).WillByDefault(ReturnRef(kMobileLanguage));
+  ON_CALL(*mock_app, ui_language()).WillByDefault(ReturnRef(kMobileLanguage));
+  ON_CALL(mock_policy_handler_, GetAppRequestTypes(_))
+      .WillByDefault(Return(std::vector<std::string>()));
+  ON_CALL(mock_policy_handler_, GetUserConsentForDevice(_))
+      .WillByDefault(Return(policy::DeviceConsent::kDeviceAllowed));
+  ON_CALL(app_mngr_, GetDeviceTransportType(_))
+      .WillByDefault(Return(hmi_apis::Common_TransportType::WIFI));
+  ON_CALL(app_mngr_, ManageHMICommand(_)).WillByDefault(Return(true));
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::SUCCESS), _));
+  EXPECT_CALL(mock_message_helper_,
+              SendChangeRegistrationRequestToHMI(testing::Eq(mock_app), _));
+  command_->Run();
+}
```

Comment:
@LevchenkoS Those two tests contain large sections of duplicated code.  The only differences are `(*msg_)[am::strings::msg_params][am::strings::app_hmi_type][0] = app_hmi_type;` addition and `ON_CALL(mock_policy_handler_, GetInitialAppData(_, _, _)) .WillByDefault(Return(true));` instead of `ON_CALL(mock_policy_handler_, GetInitialAppData(_, _, _)) .WillByDefault(DoAll(SetHmiType(app_hmi_type_str), Return(true)));` I would propose you to refactor this code in a separate method.